app/src/views/private/components/drawer-files.vue

```diff
@@ -0,0 +1,257 @@
+<template>
+	<component
+		:is="layoutWrapper"
+		v-slot="{ layoutState }"
+		v-model:selection="layoutSelection"
+		v-model:layout-options="localOptions"
+		v-model:layout-query="localQuery"
+		:filter="mergeFilters(mergeFilters(folderTypeFilter, filter), presetFilter)"
+		:filter-user="presetFilter"
+		:filter-system="mergeFilters(folderTypeFilter, filter)"
+		:search="search"
+		:collection="collection"
+		select-mode
+		:show-select="multiple ? 'multiple' : 'one'"
+	>
+		<v-drawer
+			v-model="internalActive"
+			:title="t('select_item')"
+			:small-header="currentLayout?.smallHeader"
+			:header-shadow="currentLayout?.headerShadow"
+			sidebar-resizeable
+			:sidebar-label="t('folders')"
+			@cancel="cancel"
+		>
+			<template v-if="!folder" #sidebar>
+				<files-navigation
+					:custom-target-handler="onFolderChange"
+					:current-folder="currentFolder"
+					:current-special="currentSpecial"
+					reset-open-folders
+					actions-disabled
+				/>
+			</template>
+
+			<template #navigation>
+				<v-breadcrumb :items="[{ name: collectionInfo.name, disabled: true }]" />
+			</template>
+
+			<template #subtitle>
+				<v-breadcrumb :items="[{ name: collectionInfo.name, disabled: true }]" />
+			</template>
+
+			<template #title-outer:prepend>
+				<v-button class="header-icon" rounded icon secondary disabled>
+					<v-icon :name="collectionInfo.icon" :color="collectionInfo.color" />
+				</v-button>
+			</template>
+
+			<template #actions:prepend><component :is="`layout-actions-${localLayout}`" v-bind="layoutState" /></template>
+
+			<template #actions>
+				<search-input v-model="search" v-model:filter="presetFilter" :collection="collection" />
+
+				<v-button v-tooltip.bottom="t('save')" icon rounded @click="save">
+					<v-icon name="check" />
+				</v-button>
+			</template>
+
+			<div class="layout">
+				<component :is="`layout-${localLayout}`" v-bind="layoutState">
+					<template #no-results>
+						<v-info :title="t('item_count', 0)" :icon="collectionInfo.icon" center />
+					</template>
+
+					<template #no-items>
+						<v-info :title="t('item_count', 0)" :icon="collectionInfo.icon" center />
+					</template>
+				</component>
+			</div>
+		</v-drawer>
+	</component>
+</template>
+
+<script lang="ts">
```

Comment:
With #18293 in mind, let's use script setup for new components going forward as well 👍